pkg/pillar/cmd/volumemgr/blob.go

```diff
@@ -219,20 +420,33 @@ func getBlobChildren(blob *types.BlobStatus) []*types.BlobStatus {
 			return nil
 		}
 		log.Infof("getBlobChildren(%s): adding manifest %s", blob.Sha256, manifest.Digest.Hex)
-		return []*types.BlobStatus{
-			{
-				DatastoreID: blob.DatastoreID,
-				RelativeURL: replaceSha(blob.RelativeURL, manifest.Digest),
-				Sha256:      strings.ToLower(manifest.Digest.Hex),
-				Size:        uint64(manifest.Size),
-				State:       types.INITIAL,
-				BlobType:    types.BlobManifest,
-			},
+		childHash := strings.ToLower(manifest.Digest.Hex)
+		//Check if childBlob already exists
+		existingChild := lookupOrCreateBlobStatus(ctx, sv, childHash)
+		if existingChild == nil {
+			return []*types.BlobStatus{
+				{
+					DatastoreID: blob.DatastoreID,
+					RelativeURL: replaceSha(blob.RelativeURL, manifest.Digest),
+					Sha256:      strings.ToLower(manifest.Digest.Hex),
+					Size:        uint64(manifest.Size),
+					State:       types.INITIAL,
+					BlobType:    types.BlobManifest,
+				},
+			}
+		} else if existingChild.IsLoaded {
+			// Need to update DatastoreID and RelativeURL if the blob is already loaded into CAS,
+			// because if any child blob is not downloaded already, then we would need the below data.
+			existingChild.DatastoreID = blob.DatastoreID
```

Comment:
I don't understand this part.

* Why would be need to update the datastore ID in the child? It is perfectly find for a blob to be available and not have the same datastore
* Why would we need to update the relative URL? Again, it is the verified content we care about.

Reply:
This is done only for root blob and the reason is while testing out the changes I faced the following error:
Device already had the manifest blob on boot so volumeMgr created a BlobStatus for that blob. There was 1 child blob yet to be downloaded so when handling that child blob, volumeMgr takes  DatastoreID and RelativeURL from the parent/root blob to download the child blob.
Since the rootBlob was already loaded on boot, DatastoreID and RelativeURL were not populated because of which downloader couldn’t download the missing child.